server/src/server.common.ts

```diff
@@ -192,112 +128,75 @@ export function startServer(connection: _Connection) {
         range: { start, end },
       });
 
-      connection.sendDiagnostics({ uri: textDocument.uri, diagnostics });
-      return;
+      return { diagnostics };
     }
 
-    // Basic syntax errors
-    for (const err of yamlDoc.errors) {
-      diagnostics.push({ message: err.message, range: rangeFromLinePos(err.linePos) });
-    }
+    try {
+      // Basic syntax errors
+      for (const err of yamlDoc.errors) {
+        diagnostics.push({ message: err.message, range: rangeFromLinePos(err.linePos) });
+      }
+
+      // Parse model field
+      if (yamlDoc.has("model")) {
+        diagnostics.push(...parseYamlModel(yamlDoc, lineCounter));
+        diagnostics.push(...validateYamlModel(yamlDoc.get("model") as string, yamlDoc, textDocument, map));
+      } else if (yamlDoc.has("model_file")) {
+        let position: { line: number; col: number } = { line: 0, col: 0 };
```

Comment:
Any way we can extract the part in the `if (yamlDoc.has("model_file")) {` into it's own function?

Reply:
Will clean up.